src/message.c

```diff
@@ -1112,6 +1112,8 @@ qd_message_t *qd_message(void)
 
     ZERO(msg->content);
     sys_mutex_init(&msg->content->lock);
+    sys_mutex_init(&msg->content->producer_lock);
```

Comment:
There should be corresponding sys_mutex_free() calls for the new locks in qd_message_free() where the content is freed.